Issue283 bisect deletes fpic

mikebentley15 opened the pull request:
Fixes #283 

**Description:**
FLiT Bisect with the `--delete` flag calls `make bisect-smallclean` on the bisect Makefile.  This was deleting the `bisect-XX/obj` directory, including the compiled fPIC objects.  To solve this, I

- split up the `obj/` directory into three subdirectories: `obj/fpic`, `obj/symbols`, and `obj/split`
- `obj/split` is deleted with `bisect-smallclean`
- `obj/fpic` is deleted with `bisect-clean`
- `obj/symbols` is deleted with `bisect-distclean`

**Documentation:**
No documentation changed.  The bisect directory structure is not found in the documentation.

**Tests:**
I added a test for `bisect-smallclean` in `tests/flit_cli/tst_bisect_autosqlite_clang.py` that sees if the `obj/` directory does not get deleted and that `obj/fpic` does not get deleted.

IanBriggs commented:
When I run the tests locally the only test failures I get are from the line number inconsistency problem.

mikebentley15 commented:
I'll look into it.  Thanks

- Mike

On Mon, Jul 22, 2019 at 7:47 AM Ian Briggs <notifications@github.com> wrote:

> When I run the tests locally the only test failures I get are from the
> line number inconsistency problem.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/PRUNERS/FLiT/pull/284?email_source=notifications&email_token=ACBVGNEC5PJFPN7DTJ2K7BTQAW26DA5CNFSM4IBVKVM2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2P7BGA#issuecomment-513798296>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACBVGNEZB6532X6LZKKMDT3QAW26DANCNFSM4IBVKVMQ>
> .
>

mikebentley15 commented:
I was getting other failures.  It turns out when you did the merge of `data/Makefile_bisect_binary.in`, the indentation was using spaces instead of tabs on lines 376 and 392.  I just replaced those with tabs and everything started working again.

However, when I was manually testing this out, I found another bug that I just fixed in this branch.  The `flit bisect --directory ...` command was failing if the directory that was given was a relative path instead of an absolute path.  A few changes in the source and tests needed to be made to account for this case.  It is now tested and passes.  If you think that change is too far outside of the scope of this issue, then I'm fine creating a new issue and performing a separate pull request.  It was small enough that I think it would be fine to pull it in with this one (unless you disagree).
